### DISCUSSION TITLE
[Extensions API] What is the recommended way to validate configuration options?

### DISCUSSION THREAD

**Discussion by ckrybus** [Ulauncher/Ulauncher#889] [Extensions and Themes]
I've build a very basic journal extension https://github.com/ckrybus/ulauncher-journal.

I would like to allow the user to specify a datetime format e.g. `"%d.%m.%Y %H:%M"`. When the user enters an invalid format I'd like to show a validation error. Is this currently possible?

Right now, as a workaround, I'm using `"type": "select"` in manifest.json.

**Comment by friday** [comment]
Very late answer: You can listen to http://docs.ulauncher.io/en/latest/extensions/events.html#preferencesupdateevent and show a warning with `Notify`. I'm doing this in https://github.com/friday/ulauncher-clipboard when users are switching Clipboard manager to verify that the one they're switching to exists.
